src/skmatter/_selection.py

```diff
@@ -1038,7 +1038,14 @@ def _init_greedy_search(self, X, y, n_to_select):
         self.hausdorff_ = np.full(X.shape[self._axis], np.inf)
         self.hausdorff_at_select_ = np.full(X.shape[self._axis], np.inf)
 
-        if self.initialize == "random":
+        if isinstance(self.initialize, np.ndarray):
+            if all(isinstance(i, numbers.Integral) for i in self.initialize):
+                for i, val in enumerate(self.initialize):
+                    self.selected_idx_[i] = val
+                    self._update_post_selection(X, y, self.selected_idx_[i])
+            else:
+                raise ValueError("Initialize parameter must contain only int")
```

Comment:
Would you mind adding a test for this error raise?

You can take a look how to test that in the [pytest docs](https://docs.pytest.org/en/stable/reference/reference.html#pytest.raises) or find another example in our test suite

Reply:
Just added a test for the error raise based on the other tests in test_feature_simple_fps.py. Let me know if this looks good to you!

Reply:
Should be fixed now. Added the unit tests I wrote in test_feature_simple_fps to test_sample_simple_fps. Let me know if this looks correct!